src/components/ui/frame.tsx

```diff
@@ -0,0 +1,58 @@
+"use client";
+import * as React from "react"
+import Image from "next/image";
+import type { ImgHTMLAttributes } from "react";
+import {
+    FrameUI,
+    fallbackFrameContext,
+} from "@frames.js/render";
+import { signFrameAction, FarcasterSigner } from '@frames.js/render/farcaster'
+import { useFrame } from "@frames.js/render/use-frame";
+import {AccountObjectType} from "@/stores/useAccountStore";
+
+export function FrameImageNext(
+    props: ImgHTMLAttributes<HTMLImageElement> & { src: string }
+): React.JSX.Element {
+    return (
+        <Image
+            {...props}
+            alt={props.alt ?? ""}
+            sizes="100vw"
+            height={0}
+            width={0}
+        />
+    );
+}
+
+
+export default function Frame( { homeFrameUrl, account } : { homeFrameUrl: string, account?: AccountObjectType }) {
+    const farcasterSigner: FarcasterSigner | undefined = account ? {
```

Comment:
@hellno I was not able to test this due do an endpoint error, under `/accounts` and __Connect__, which might be easier to sort out for you:

```json
{
    "code": "42501",
    "details": null,
    "hint": null,
    "message": "permission denied for function crypto_aead_det_encrypt"
}

Reply:
I've seen this error before but have no immediate fix right now.
one step forward might be to run this sql script in the supabase local instance at `https://localhost:54323/`

```sql
GRANT
EXECUTE
  ON FUNCTION pgsodium.crypto_aead_det_encrypt (bytea, bytea, bytea, bytea) TO authenticated;

GRANT
EXECUTE
  ON FUNCTION pgsodium.crypto_aead_det_encrypt (bytea, bytea, uuid, bytea) TO authenticated;

GRANT
EXECUTE
  ON FUNCTION pgsodium.crypto_aead_det_decrypt (bytea, bytea, bytea, bytea) TO authenticated;

GRANT
EXECUTE
  ON FUNCTION pgsodium.crypto_aead_det_decrypt (bytea, bytea, uuid, bytea) TO authenticated;

GRANT
  pgsodium_keyiduser TO authenticated;
```